### ISSUE TITLE
Expo EAS

### ISSUE THREAD

**Issue by craigmulligan** [homeeondemand/react-native-mapbox-navigation#112]
Has anyone had a go at packaging this with expo's new [EAS build system](https://docs.expo.dev/build/introduction/)? I'm planning on working on it but checking if anyone has got it working before. 

As far as I can tell it requires writing a expo plugin that writes the manifest files for each platform at build time. You can see an example of bluetooth here: https://github.com/expo/config-plugins/tree/main/packages/react-native-ble-plx

**Comment by mckeny3** [comment]
did you manage to get it done?

**Comment by craigmulligan** [comment]
I haven't yet. But I do still plan on working on it. Will update that issue when I get around to it.

**Comment by zivc** [comment]
Any updates @craigmulligan? I await your feedback with baited breath :S

**Comment by craigmulligan** [comment]
Just a small update, I had a quick look and it appears the [nmapbox packages plugin](https://github.com/rnmapbox/maps/blob/main/plugin/src/withMapbox.ts) does everything we need for android, so just installing it along with this package works fine, the problem now is that react-native-mapbox-navigation uses and older version of gradle than the expo build system so it's failing to build because of deprecations in gradle 7 so trying to figure that out atm.

**Comment by ismael-su** [comment]
@craigmulligan  i'm going through the same issue regarding old gradle version, have you managed to make it work ?

**Comment by craigmulligan** [comment]
No haven't looked at it again. I still hope to at some point.

**Comment by JKKholmatov** [comment]
Any updates?

**Comment by ismael-su** [comment]
> Any updates?

Am working on a workaround. Works with native module on Android but eventually crashes when route tiles render. By not showing the route tiles it works
I will soon publish the npm package

**Comment by mckeny3** [comment]
yep...gave me headache...i had to switch to flutter...the mapbox navigation work there

**Comment by santitopo** [comment]
> > Any updates?
> 
> Am working on a workaround. Works with native module on Android but eventually crashes when route tiles render. By not showing the route tiles it works
> I will soon publish the npm package 

Hey @ismael-su ! I'm trying to implement an Expo Module myself to use the latest versions of the iOS and Android Mapbox Navigation sdks (so basically skipping this library which seems unmantained), and I came across your comment.

I currently have it working in iOS but in Android it crashes as soon as the map attempts to render. Im trying to instantiate a very basic NavigationView (btw, the same code does work in a native Android project).

Can you tell me more about how you solved it if you happen to do it? and also how can I disable the tiles so I can see if at least that gets me through instantiating the view.

Thanks in advance!

**Comment by KindCoder-no** [comment]
Did anyone manage to get it working with expo eas?

**Comment by ismael-su** [comment]
@santitopo  hello, just realized you mentionned me, i stopped working on the project months ago because i wasn't even able to reproduce the working build. am still waiting

**Comment by cmaycumber** [comment]
I'm thinking about creating an expo-plugin. Wondering if I need to use the forked library from #116 or if this is the repo I should attempt to do it on. 

Does anyone have feedback on what library they're currently using?

**Comment by BryanFonseca** [comment]
In case someone else is still struggling with this library, I managed to get it to work on EAS with this one: https://github.com/pawan-pk/react-native-mapbox-navigation
